From f24479ae506fdf228d0ee5b592b6c0256b7b8bea Mon Sep 17 00:00:00 2001
From: Bo Lv <bo.lv@amlogic.com>
Date: Thu, 29 Jul 2021 10:25:24 +0800
Subject: [PATCH 2/2] bl33/v2019: fix compile error when using higher version
 of GCC [1/1]

PD#SWPL-55750

Problem:
compile error when using high version of GCC.

Solution:
fix it.

Verify:
compile pass on all boards

Change-Id: I8336f372b5de817cc5c80b0e1a030c943a0b275e
Signed-off-by: Bo Lv <bo.lv@amlogic.com>
---
 arch/arm/include/asm/arch-a1/ddr_define.h          |  2 +-
 arch/arm/include/asm/arch-c1/ddr_define.h          |  2 +-
 arch/arm/include/asm/arch-sc2/ddr_define.h         |  2 +-
 .../arm/mach-meson/g12a/firmware/scp_task/Makefile | 14 +++++++-------
 .../arm/mach-meson/g12b/firmware/scp_task/Makefile | 14 +++++++-------
 board/amlogic/s4_ap201/firmware/timing.c           |  2 +-
 board/amlogic/s4_ap222/firmware/timing.c           |  2 +-
 board/amlogic/s4_ap223/firmware/timing.c           |  2 +-
 board/amlogic/s4_ap229/firmware/timing.c           |  2 +-
 board/amlogic/s4_ap232/firmware/timing.c           |  2 +-
 board/amlogic/s4_aq222/firmware/timing.c           |  2 +-
 board/amlogic/s4_aq229/firmware/timing.c           |  2 +-
 board/amlogic/s4_skt/firmware/timing.c             |  2 +-
 board/amlogic/s4_t211/firmware/timing.c            |  2 +-
 drivers/clk/meson/clk_meson_a1.c                   |  2 +-
 15 files changed, 27 insertions(+), 27 deletions(-)

diff --git a/arch/arm/include/asm/arch-a1/ddr_define.h b/arch/arm/include/asm/arch-a1/ddr_define.h
index 2d2697e498..929c3e7ed8 100644
--- a/arch/arm/include/asm/arch-a1/ddr_define.h
+++ b/arch/arm/include/asm/arch-a1/ddr_define.h
@@ -261,7 +261,7 @@
 											DDR_FUNC_LPDDR3_CA_TRAINING_CA1_BIT0| \
 											DDR_FUNC_LPDDR3_CA_TRAINING_CA1_BIT1| \
 											DDR_FUNC_LPDDR3_SOC_ODT_ONLY_UP	| \
-											(1 << 31) 						\
+											(1u << 31) 						\
 											)
 
 
diff --git a/arch/arm/include/asm/arch-c1/ddr_define.h b/arch/arm/include/asm/arch-c1/ddr_define.h
index 2d2697e498..929c3e7ed8 100644
--- a/arch/arm/include/asm/arch-c1/ddr_define.h
+++ b/arch/arm/include/asm/arch-c1/ddr_define.h
@@ -261,7 +261,7 @@
 											DDR_FUNC_LPDDR3_CA_TRAINING_CA1_BIT0| \
 											DDR_FUNC_LPDDR3_CA_TRAINING_CA1_BIT1| \
 											DDR_FUNC_LPDDR3_SOC_ODT_ONLY_UP	| \
-											(1 << 31) 						\
+											(1u << 31) 						\
 											)
 
 
diff --git a/arch/arm/include/asm/arch-sc2/ddr_define.h b/arch/arm/include/asm/arch-sc2/ddr_define.h
index 2d2697e498..929c3e7ed8 100644
--- a/arch/arm/include/asm/arch-sc2/ddr_define.h
+++ b/arch/arm/include/asm/arch-sc2/ddr_define.h
@@ -261,7 +261,7 @@
 											DDR_FUNC_LPDDR3_CA_TRAINING_CA1_BIT0| \
 											DDR_FUNC_LPDDR3_CA_TRAINING_CA1_BIT1| \
 											DDR_FUNC_LPDDR3_SOC_ODT_ONLY_UP	| \
-											(1 << 31) 						\
+											(1u << 31) 						\
 											)
 
 
diff --git a/arch/arm/mach-meson/g12a/firmware/scp_task/Makefile b/arch/arm/mach-meson/g12a/firmware/scp_task/Makefile
index ecc4671be6..26d84be7c3 100644
--- a/arch/arm/mach-meson/g12a/firmware/scp_task/Makefile
+++ b/arch/arm/mach-meson/g12a/firmware/scp_task/Makefile
@@ -6,13 +6,13 @@ include $(buildtree)/include/autoconf.mk
 include $(buildtree)/.config
 
 # Select ARMv7-m bare-metal toolchain
-CROSS_COMPILE=arm-none-eabi-
-ASM=$(CROSS_COMPILE)as
-CC=$(CROSS_COMPILE)gcc
-CPP=$(CROSS_COMPILE)cpp
-LD=$(CROSS_COMPILE)ld
-OBJCOPY=$(CROSS_COMPILE)objcopy
-OBJDUMP=$(CROSS_COMPILE)objdump
+CROSS_COMPILE_BL301 ?=arm-none-eabi-
+ASM=$(CROSS_COMPILE_BL301)as
+CC=$(CROSS_COMPILE_BL301)gcc
+CPP=$(CROSS_COMPILE_BL301)cpp
+LD=$(CROSS_COMPILE_BL301)ld
+OBJCOPY=$(CROSS_COMPILE_BL301)objcopy
+OBJDUMP=$(CROSS_COMPILE_BL301)objdump
 
 ARM_ARCH=armv7-m
 ARM_CPU=cortex-m3
diff --git a/arch/arm/mach-meson/g12b/firmware/scp_task/Makefile b/arch/arm/mach-meson/g12b/firmware/scp_task/Makefile
index ecc4671be6..26d84be7c3 100644
--- a/arch/arm/mach-meson/g12b/firmware/scp_task/Makefile
+++ b/arch/arm/mach-meson/g12b/firmware/scp_task/Makefile
@@ -6,13 +6,13 @@ include $(buildtree)/include/autoconf.mk
 include $(buildtree)/.config
 
 # Select ARMv7-m bare-metal toolchain
-CROSS_COMPILE=arm-none-eabi-
-ASM=$(CROSS_COMPILE)as
-CC=$(CROSS_COMPILE)gcc
-CPP=$(CROSS_COMPILE)cpp
-LD=$(CROSS_COMPILE)ld
-OBJCOPY=$(CROSS_COMPILE)objcopy
-OBJDUMP=$(CROSS_COMPILE)objdump
+CROSS_COMPILE_BL301 ?=arm-none-eabi-
+ASM=$(CROSS_COMPILE_BL301)as
+CC=$(CROSS_COMPILE_BL301)gcc
+CPP=$(CROSS_COMPILE_BL301)cpp
+LD=$(CROSS_COMPILE_BL301)ld
+OBJCOPY=$(CROSS_COMPILE_BL301)objcopy
+OBJDUMP=$(CROSS_COMPILE_BL301)objdump
 
 ARM_ARCH=armv7-m
 ARM_CPU=cortex-m3
diff --git a/board/amlogic/s4_ap201/firmware/timing.c b/board/amlogic/s4_ap201/firmware/timing.c
index a0c83e7b3b..a22b61f530 100644
--- a/board/amlogic/s4_ap201/firmware/timing.c
+++ b/board/amlogic/s4_ap201/firmware/timing.c
@@ -28,7 +28,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_ap222/firmware/timing.c b/board/amlogic/s4_ap222/firmware/timing.c
index 5d7f7be634..7266f88b97 100644
--- a/board/amlogic/s4_ap222/firmware/timing.c
+++ b/board/amlogic/s4_ap222/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_ap223/firmware/timing.c b/board/amlogic/s4_ap223/firmware/timing.c
index 94d011d6e6..fdead4e234 100644
--- a/board/amlogic/s4_ap223/firmware/timing.c
+++ b/board/amlogic/s4_ap223/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_ap229/firmware/timing.c b/board/amlogic/s4_ap229/firmware/timing.c
index c217b7e995..1ba8f1f38a 100644
--- a/board/amlogic/s4_ap229/firmware/timing.c
+++ b/board/amlogic/s4_ap229/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_ap232/firmware/timing.c b/board/amlogic/s4_ap232/firmware/timing.c
index f22ee89497..47c964dc71 100644
--- a/board/amlogic/s4_ap232/firmware/timing.c
+++ b/board/amlogic/s4_ap232/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_aq222/firmware/timing.c b/board/amlogic/s4_aq222/firmware/timing.c
index 338d0c1f04..6906fa4bee 100644
--- a/board/amlogic/s4_aq222/firmware/timing.c
+++ b/board/amlogic/s4_aq222/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_aq229/firmware/timing.c b/board/amlogic/s4_aq229/firmware/timing.c
index f738214596..1f98bcdbc4 100644
--- a/board/amlogic/s4_aq229/firmware/timing.c
+++ b/board/amlogic/s4_aq229/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_skt/firmware/timing.c b/board/amlogic/s4_skt/firmware/timing.c
index 11060dd53b..0514372518 100644
--- a/board/amlogic/s4_skt/firmware/timing.c
+++ b/board/amlogic/s4_skt/firmware/timing.c
@@ -27,7 +27,7 @@
 
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
-#define DDR_ID_FROM_ADC  (0Xfe<<24)
+#define DDR_ID_FROM_ADC  (0Xfeu<<24)
 #define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
diff --git a/board/amlogic/s4_t211/firmware/timing.c b/board/amlogic/s4_t211/firmware/timing.c
index d12ab52a00..39cea245ff 100644
--- a/board/amlogic/s4_t211/firmware/timing.c
+++ b/board/amlogic/s4_t211/firmware/timing.c
@@ -28,7 +28,7 @@
 //bit24-31 define ID and size
 #define DDR_ID_FROM_EFUSE  (0Xff<<24)
 #define DDR_ID_FROM_ADC  (0Xfe<<24)
-#define DDR_ID_FROM_GPIO_CONFIG1  (0Xfd<<24)
+#define DDR_ID_FROM_GPIO_CONFIG1  (0Xfdu<<24)
 #define DDR_ID_START_MASK  (0XFFDDCCBB)
 
 #define DDR_ADC_CH0  (0X0<<6)
diff --git a/drivers/clk/meson/clk_meson_a1.c b/drivers/clk/meson/clk_meson_a1.c
index 83b155ce78..0ed4687ce8 100644
--- a/drivers/clk/meson/clk_meson_a1.c
+++ b/drivers/clk/meson/clk_meson_a1.c
@@ -158,7 +158,7 @@ static ulong meson_clk_set_rate(struct clk *clk, ulong rate)
 	unsigned int div_val;
 	struct meson_clk *priv = dev_get_priv(clk->dev);
 	unsigned int i;
-	struct meson_div *div;
+	struct meson_div *div = NULL;
 
 	for (i = 0; i < ARRAY_SIZE(divs); i++) {
 		if (clk->id == divs[i].index)
-- 
2.29.2

